action.php

```diff
@@ -312,14 +366,51 @@ private function showPageWithErrorMsg(Doku_Event $event, $msglangkey, $replaceme
     }
 
     /**
+     * Returns the parsed Wikitext in dw2pdf for the given id and revision
+     *
+     * @param string     $id  page id
+     * @param string|int $rev revision timestamp or empty string
+     * @param string     $date_at
+     * @return null|string
+     */
+    protected function p_wiki_dw2pdf($id, $rev = '', $date_at = '') {
+        $file = wikiFN($id, $rev);
+
+        if(!file_exists($file)) return '';
+
+        //ensure $id is in global $ID (needed for parsing)
+        global $ID;
+        $keep = $ID;
+        $ID   = $id;
+
+        $ret  = '';
+
+        if($rev || $date_at) {
+            $ret = p_render('dw2pdf', p_get_instructions(io_readWikiPage($file, $id, $rev)), $info, $date_at); //no caching on old revisions
+        } else {
+            $ret = p_cached_output($file, 'dw2pdf', $id);
+        }
+
+        //restore ID (just in case)
+        $ID = $keep;
+
+        return $ret;
+    }
+
+    /**
      * Build a pdf from the html
      *
      * @param string $cachefile
      */
     protected function generatePDF($cachefile) {
-        global $ID;
-        global $REV;
-        global $INPUT;
+        global $ID, $REV, $INPUT, $DATE_AT, $ACT;
+
+        if ($ACT == 'export_pdf') { //only one page is exported
+            $rev = $REV;
+            $date_at = $DATE_AT;
+        } else { //we are exporting entre namespace, ommit revisions
+            $rev = $date_at = '';
+        }
```

Comment:
So it could happen, that a revision was given initially but it's later discarded here. However at that point the decision to use a temporary file was made already. Is that correct? Do we cache namespace exports?

Reply:
This is an edge case, but I reimplemented caching under these circumstances.